nncf/experimental/common/pruning/operations.py

```diff
@@ -344,38 +354,36 @@ def mask_propagation(
         cls, node: NNCFNode, graph: NNCFGraph, tensor_processor: Type[NNCFPruningBaseTensorProcessor]
     ) -> None:
         input_masks = get_input_masks(node, graph)
-        if not input_masks:
-            input_masks = [None]
-
-        assert len(input_masks) == 1
-        output_mask = input_masks[0]
-
-        assert isinstance(node.layer_attributes, MultipleOutputLayerAttributes)
-        chunk_axis = node.layer_attributes.axis
-        chunks = node.layer_attributes.chunks
-
-        input_edge = graph.get_input_edges(node)[0]
-        output_edge = graph.get_output_edges(node)[0]
-        input_shape = input_edge.tensor_shape
-        output_shape = output_edge.tensor_shape
-        is_chunk_axis_removed = (chunks == input_shape[chunk_axis]) and (len(input_shape) > len(output_shape))
-        is_unit_chunk = input_shape[chunk_axis] // chunks == 1
-        if is_chunk_axis_removed or is_unit_chunk:
-            output_mask = PropagationMask()
-            for dim, groups in input_masks[0].dim_groups_map.items():
-                if dim != chunk_axis:
-                    # not affected by split groups are propagated further
-                    output_mask.dim_groups_map[dim] = groups
-                else:
-                    # invalidate groups, that assigned to the removed dimension or to the dimension that have 1 channel
-                    for group in groups:
-                        group.invalidate()
-        else:
-            nncf_logger.warning(
-                "symbolic mask propagation for split by prune dimension is not implemented, "
-                "just propagate further for now"
-            )
+        output_mask = None
+        if input_masks:
+            assert len(input_masks) == 1
+            assert isinstance(node.layer_attributes, MultipleOutputLayerAttributes)
```

Comment:
This if making code hard to read, its body is too big IMHO, could you please create auxiliary function?